src/basic/utf8.c

```diff
@@ -389,11 +389,23 @@ char *utf16_to_utf8(const char16_t *s, size_t length /* bytes! */) {
         const uint8_t *f;
         char *r, *t;
 
+        if (length == 0)
+                return new0(char, 1);
+
         assert(s);
 
+        if (length == SIZE_MAX) {
+                length = char16_strlen(s);
+
+                if (length > SIZE_MAX/2)
+                        return NULL; /* overflow */
+
+                length *= 2;
```

Comment:
Why do you still keep doing this? `__builtin_*_overflow()` is a thing and soo much better than some random manual overflow check that is prone to get wrong. It's even self-documenting!
```suggestion
                if (__builtin_mul_overflow(length, 2, &length)
                        return NULL;
```

Reply:
we can look into using that more, but we generally try to avoid __builtin stuff without macro glue.

I must say I find it hard to read with the builtin though. And sometimes we can do multiple overflow checks at once, like the other one in this patch here